src/server/qtquick/private/wquicktextinputv1.cpp

```diff
@@ -0,0 +1,575 @@
+#include "wquicktextinputv1_p.h"
+#include "wseat.h"
+#include "wsurface.h"
+#include "wtools.h"
+
+#include <qwdisplay.h>
+#include <qwsignalconnector.h>
+#include <qwseat.h>
+#include <qwcompositor.h>
+
+#include <QRect>
+
+#include <text-input-unstable-v1-protocol.h>
+#include <wayland-server-protocol.h>
+#include <wayland-server-core.h>
+#include <wayland-server.h>
+
+extern "C" {
+#include <wlr/types/wlr_seat.h>
+#define static
+#include <wlr/types/wlr_compositor.h>
+#undef static
+#include <wlr/util/box.h>
+}
+
+QW_USE_NAMESPACE
+WAYLIB_SERVER_BEGIN_NAMESPACE
+struct ws_text_input_v1 {
```

Comment:
What is it?

Reply:
Native handle defined by those who implement the protocol, like wlr_text_input_v3.